compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -905,16 +905,19 @@ object RefChecks {
       owner.isDeprecated
       || isEnumOwner(owner)
 
+    def isDeprecatedOrSyntheticMethod(owner: Symbol) = owner.isDeprecated || (owner.isRealMethod && owner.is(Synthetic))
```

Comment:
a call to this method should replace the first condition of `isDeprecatedOrEnum`